src/komponenter/StatusIkon/StatusIkon.tsx

```diff
@@ -6,15 +6,27 @@ import { ReactComponent as ProblemIkon } from '@/assets/ikoner/varsel.svg';
 import { ReactComponent as InaktivIkon } from '@/assets/ikoner/inaktiv.svg';
 import { ReactComponent as AvbruttIkon } from '@/assets/ikoner/stop.svg';
 import { ReactComponent as GjennomforesIkon } from '@/assets/ikoner/gjennomfores.svg';
+import { Rolle } from '@/AvtaleContext';
 
-const StatusIkon: FunctionComponent<{ status: string } & React.SVGProps<SVGSVGElement>> = props => {
+const StatusIkon: FunctionComponent<{
+    status: string;
+    rolle?: Rolle;
+    godkjentAvInnloggetBruker?: boolean;
+} & React.SVGProps<SVGSVGElement>> = props => {
     switch (props.status) {
         case 'Påbegynt':
             return <PabegyntIkon {...props} />;
         case 'Klar for oppstart':
             return <CheckIkon {...props} />;
         case 'Mangler godkjenning':
-            return <ProblemIkon {...props} />;
+            if (
+                (props.rolle !== undefined && props.rolle === 'VEILEDER') ||
+                (props.godkjentAvInnloggetBruker !== undefined && props.godkjentAvInnloggetBruker)
```

Comment:
Trenger ikke sjekke for undefined. Kan skrive `props.rolle === 'VEILEDER' || props.godkjentAvInnloggetBruker`